src/cargo/core/resolver/mod.rs

```diff
@@ -823,15 +823,12 @@ fn generalize_conflicting(
     dep: &Dependency,
     conflicting_activations: &ConflictMap,
 ) -> Option<ConflictMap> {
-    if conflicting_activations.is_empty() {
-        return None;
-    }
     // We need to determine the `ContextAge` that this `conflicting_activations` will jump to, and why.
-    let (backtrack_critical_age, backtrack_critical_id) = conflicting_activations
-        .keys()
-        .map(|&c| (cx.is_active(c).expect("not currently active!?"), c))
-        .max()
-        .unwrap();
+    let (backtrack_critical_age, backtrack_critical_id) = shortcircuit_max(
```

Comment:
Given we've changed this. Does [this line](https://github.com/rust-lang/cargo/pull/11770/files#diff-ebf960f030fe60e669bff9c364f8f8c44fda2fe78c6bbaa8858f69c0ba5b15eaR817) still hold?

```
/// Panics if the input conflict is not all active in `cx`.
```